source/write/bulk-write.txt

```diff
@@ -0,0 +1,287 @@
+to perform. A query filter specifies the criteria used to match documents to
+in your collection.
+
+``UpdateOne`` updates *the first* document that matches your query filter.
+``UpdateMany`` updates *all* documents that match your query filter.
+
+The following example creates an instance of ``UpdateOne``:
+
+.. literalinclude:: /includes/write/bulk-write.py
+   :start-after: start-bulk-update-one
+   :end-before: end-bulk-update-one
+   :language: python
+   :copyable:
+
+The following example creates an instance of ``UpdateMany``:
+
+.. literalinclude:: /includes/write/bulk-write.py
+   :start-after: start-bulk-update-many
+   :end-before: end-bulk-update-many
+   :language: python
+   :copyable:
+
+Replace Operations
+~~~~~~~~~~~~~~~~~~
+
+Replace operations remove all fields and values of a specified document and
+replaces them with new ones. To perform a replace operation, create an instance
+of ``ReplaceOne`` and pass it a query filter and the fields and values you want
+to replace with. To replace multiple documents, you must create
+an instance of ``ReplaceOne`` for each document.
+
+The following example creates an instance of ``ReplaceOne``:
+
+.. literalinclude:: /includes/write/bulk-write.py
+   :start-after: start-bulk-replace-one
+   :end-before: end-bulk-replace-one
+   :language: python
+   :copyable:
+
+Delete Operations
+~~~~~~~~~~~~~~~~~
+
+To perform a delete operation, create an instance of ``DeleteOne`` or ``DeleteMany`` and pass it a
+query filter specifying the documents you want to delete.
+
+``DeleteOne`` removes *the first* document that matches your query filter.
+``DeleteMany`` removes *all* documents that match your query filter.
+
+The following example creates an instance of ``DeleteOne``:
+
+.. literalinclude:: /includes/write/bulk-write.py
+   :start-after: start-bulk-delete-one
+   :end-before: end-bulk-delete-one
+   :language: python
+   :copyable:
+
+The following example creates an instance of ``DeleteMany``:
+
+.. literalinclude:: /includes/write/bulk-write.py
+   :start-after: start-bulk-delete-many
+   :end-before: end-bulk-delete-many
+   :language: python
+   :copyable:
+
+Mixed Bulk Write Operations
+----------------------------
+
+You can perform multiple types of write operations in a single ``bulk_write()``
+method call. To do so, pass a list of operation class instances to the
+``bulk_write()`` method. By default, the method runs each operation in the order
+it's provided.
+
+The following example performs multiple write operations by using the
+``bulk_write()`` method:
+
+.. literalinclude:: /includes/write/bulk-write.py
+   :start-after: start-bulk-write-mixed
+   :end-before: end-bulk-write-mixed
+   :language: python
+   :copyable:
+
+If any of the write operations fail, {+driver-short+} raises a
+``BulkWriteError`` and does not perform any further operations.
+``BulkWriteError`` provides a ``details`` attribute that includes the operation
+that failed, and details about the exception.
```

Comment:
Move to Troubleshooting section at page end?

Reply:
I put this here since it only applies to an ordered bulk write shown in this section example, not to ``bulk_write`` in general. It seems a little more informational than troubleshooting imo but I can move it if you feel strongly